src/shogun/base/manifest.cpp

```diff
@@ -0,0 +1,65 @@
+#include <shogun/base/manifest.h>
+
+#include <unordered_map>
+
+namespace shogun
+{
+
+	class Manifest::Self
+	{
+		public:
+			std::string description;
+			std::unordered_map<std::string, Any> classes;
+	};
+
+	Manifest::Manifest(const std::string& description, 
+			const std::initializer_list<std::pair<std::string,Any>> classes) :
+		self()
+	{
+		self->description = description;
+		for (const auto& m : classes)
+			addClass(m.first, m.second);
+	}
+
+	Manifest::Manifest(const Manifest& other) : 
+		self()
+	{
+		self->description = other.self->description;
+		self->classes = other.self->classes;
+	}
+
+	Manifest& Manifest::operator=(const Manifest& other)
+	{
+		self->description = other.self->description;
+		self->classes = other.self->classes;
+		return *this;
+	}
+
+	bool operator==(const Manifest& first, const Manifest& second)
+	{
+		return first.self->description == second.self->description
+			and	first.self->classes == second.self->classes;	
```

Comment:
first, what is the weird whitespace between `and` and `first.self->classes`

second i would assure parenthesis, like
```
return (first.self->description == second.self->description)
and	(first.self->classes == second.self->classes);
```

and have you checked that the conditions for `bool operator==` of `undorder_map` holds for `Any`: http://en.cppreference.com/w/cpp/container/unordered_map/operator_cmp